### PR TITLE
Added method to get data at par in RClient class

### PR DESCRIPTION
**Pull request by stanislavlyalin**
## Overview

This PR added the ability to get data at par from RNode via gRPC. The corresponding method was added in RNode in [PR](https://github.com/rchain/rchain/pull/3592). This new method allows getting deploy results faster. Please see the PRs description mentioned above and the related issue for more details.

### How to check solution

1. update pyrchain lib from source repo with command `python3 -m pip install -e .` (from repo root)
2. run RNode as validator locally
3. from client code make **deploy** with the code
```new return(`rho:rchain:deployId`) in { return!("data-at-par") }```
and then **propose**
4. call client method to get the result
```python
from rchain.client import RClient, DataQueries

host = '127.0.0.1'
client = RClient(host, 40401)

# block_hash is result of propose (hash-string)
par = DataQueries.deploy_id(sig)  # sig is deploy signature (id) as result of deploy (hash-string)
use_pre_state_hash = False

data = client.get_data_at_par(par, block_hash, use_pre_state_hash)
```
5. The sample result is
```
par {
  exprs {
    g_string: "data-at-par"
  }
}
block {
  blockHash: "b2a926c00f5f0730b8bf91a02f837ac646b09b54f362699cf051f5d8417033af"
  sender: "04415d35d799325f7784d7fbae15a3c4d0a24ff97429c7512b877b34bf3b7c801d220c38877e9a8a0736d168000ac5139cf505d9c030d1b7714dce4128ca3771b3"
  seqNum: 31
  sig: "3045022100ba08aaf1f65aa61a12715477d50beffcad946759996f47288547b4eff8f604ad02206b86c0e6c7579b7b6dd16c930cf6f15729f2a6a12e0876133cefa4df716fb9d6"
  sigAlgorithm: "secp256k1"
  shardId: "root"
  version: 1
  timestamp: 1644476891910
  parentsHashList: "aa5dababfc583d2e9c72a6ad81c71d7dea1ee9eed294513a1450a3e4e3d7cede"
  blockNumber: 31
  preStateHash: "7d0d701c4ec20c6baa8053025fa8663ea558fb09b988bf79f6a48887ea3adf74"
  postStateHash: "82718a7e01d3e1c41d25c14e02402d97ef02ca9a8af340df3c61fc1ce4db1854"
  bonds {
    validator: "04415d35d799325f7784d7fbae15a3c4d0a24ff97429c7512b877b34bf3b7c801d220c38877e9a8a0736d168000ac5139cf505d9c030d1b7714dce4128ca3771b3"
    stake: 150500000000000
  }
  blockSize: "127861"
  deployCount: 1
  faultTolerance: 1.0
  justifications {
    validator: "04415d35d799325f7784d7fbae15a3c4d0a24ff97429c7512b877b34bf3b7c801d220c38877e9a8a0736d168000ac5139cf505d9c030d1b7714dce4128ca3771b3"
    latestBlockHash: "aa5dababfc583d2e9c72a6ad81c71d7dea1ee9eed294513a1450a3e4e3d7cede"
  }
}
```

### Notes
* The method name get_data_at_par isn't match the gRPC method getDataAtName, but this name already in use for listenForDataAtName gRPC method
* Getting Par from deploy signature (```par = DataQueries.deploy_id(sig)```) can be hidden inside RClient implementation. Is it make sense?
* Is library version should be updated? It is feature adding so I guess the second number should be increased (`0.3.11-rc3 -> 0.4.0`)

### CODE REVIEW

File: rchain/client.py [hunk 1]
```diff
@@ -184,6 +184,10 @@ def get_data_at_name(self, par: Par, depth: int = -1) -> Data:
         wrapped = response.payload
         return Data.FromString(wrapped.SerializeToString())
 
+    def get_data_at_par(self, par: Par, block_hash: str, use_pre_state_hash: bool) -> RhoDataResponse:
+        query = DataAtNameByBlockQuery(par=par, blockHash=block_hash, usePreStateHash=use_pre_state_hash)
+        return self._deploy_stub.getDataAtName(query)
```
Comment (tgrospic) [inline rchain/client.py]: The check for additional errors is missing - `self._check_response`.

_The reason for this additional error checking is wrong implementation for error handling in RNode gRPC service definitions with `ServiceError` type._

Comment (stanislavlyalin) [reply rchain/client.py]: Yes, good point. Fixed in [82fa891](https://github.com/rchain/pyrchain/pull/81/commits/82fa8916f477de73528d873b5fa6328d74da6d8a).

**Review by zsluedem — APPROVED** [review]
LGTM

**Review by tgrospic — APPROVED** [review]
LGTM

**Pull request merged**